new PR after feedback from nature people (#3)
greenelab/nature_news_manuscript: content/03.methods.md

 
### Name Formatting

#### Name Formatting for Gender Prediction in Quotes or Mentions
To identify the gender of a quoted or mentioned person, we first attempt to identify the person’s full name.
Even though genderizeR only uses the first name to make the gender prediction, identifying the full name gives us greater confidence that we are using the first name.
To identify the full name, we take the predicted speaker by coreNLP and match it to the longest matching name within the same article.
We match names by finding the longest mentioned name in the article with minimal edit (Levenshtein) distance.
The name with the smallest edit distance, where character deletions have zero cost, is defined as the matching name.
Character deletion was assigned a zero cost because we would like exact substring matches.
For example, the calculated cost, including a cost for character deletion, between John and John Steinberg is 10; without character deletion, it is 0.
Compared with the distance between John and Jane Doe, with character deletion cost, it is 7; without it is 2.

> If we are still unable to find a full name, or if coreNLP cannot identify a speaker at all, we also look if coreNLP identified a gendered pronoun to the quote.

Reviewer: ```suggestion
If we are still unable to find a full name, or if coreNLP cannot identify a speaker at all, we also determine whether or not coreNLP linked a gendered pronoun to the quote.
```

> If so, we assign the gender of the speaker to the gender of the pronoun.

Reviewer: ```suggestion
If so, we assign the gender of the pronoun to the speaker.
```
I think - would be good to get eyes on this. 👀

We ignore all quotes with no name or partial names and no associated pronouns.

#### Name Formatting for Gender Prediction of Authors

> Only articles with more than one author were considered.

Reviewer: ```suggestion

Because we separate first and last authors, we only considered articles with more than one author.
```

> Similarly for the name formatting for quotes, we need to extract the first name of the authors.

Reviewer: ```suggestion
As for quotes, we needed to extract the first name of the authors.
```

> Names were cast to lowercase and processed using the R package humaniformat [@https://cran.r-project.org/web/packages/humaniformat/index.html].

Reviewer: ```suggestion
We cast names to lowercase and processed them using the R package humaniformat [@https://cran.r-project.org/web/packages/humaniformat/index.html].
```

humaniformat identifies if names are reversed (Lastname, Firstname), as well as identifies middle names.

> This processing was not done in the quotes prediction because we assume the names written in an article are not reversed or abbreviated.

Reviewer: ```suggestion
This processing was not required for quote prediction because we names written in news articles did not appear to be reversed or abbreviated.
```

Furthermore, since many articles only contain first name initials (for example, "N. Davidson"), we remove any names less than four letters (length includes punctuation) and containing a  "." or "-", then strip out all periods from the first name.
This ensures that hyphenated names are not changed, e.g. Julia-Louise remains unchanged, but removes hyphenated initials, e.g. J-L.
Finally, we only consider any remaining first names of more than two characters.
This is to eliminate first and middle jointly-initialized names.
For example, "NR Davidson" would be reduced to "Davidson" and then eliminated due to the lack of a first name.

#### Name Formatting for Name Origin Prediction
In contrast to the gender prediction, we require the entire name in all steps of name origin prediction.
For names identified in the _Nature_ news articles, we use the same process as described for the gender prediction; we again try to identify the full name.
For author names, we process the names as previously described for the gender prediction of authors.
For all names, we only consider them in our analyses if they consist of two distinct parts separated by a space.

> Additionally, if a full name is less than three characters, it is not considered.

Reviewer: ```suggestion
Additionally, if a full name is less than three characters, we were unable to consider it as the prediction model that we apply uses 3-mers.
```
I think this is a technical constraint that would make it impossible to apply without retraining - so we can note that as a reason.
